File: support/src/main/java/io/wax911/support/base/attribute/SeasonType.java
package io.wax911.support.base.attribute;

import androidx.annotation.StringDef;
import kotlin.annotation.AnnotationRetention;
import kotlin.annotation.Retention;

@Retention(AnnotationRetention.SOURCE)
@StringDef({SeasonType.WINTER, SeasonType.SPRING,
        SeasonType.SUMMER, SeasonType.FALL})
public @interface SeasonType {
    String WINTER = "WINTER",
            SPRING = "SPRING",
            SUMMER = "SUMMER",
            FALL = "FALL";

    String[] Seasons = {WINTER, SPRING, SUMMER, FALL};
}

File: support/src/main/java/io/wax911/support/base/attribute/SeasonType.kt
package io.wax911.support.base.attribute

import androidx.annotation.StringDef

@Retention(AnnotationRetention.SOURCE)
@StringDef(SeasonType.WINTER, SeasonType.SPRING, SeasonType.SUMMER, SeasonType.FALL)
annotation class SeasonType {
    companion object {
        const val WINTER = "WINTER"
        const val SPRING = "SPRING"
        const val SUMMER = "SUMMER"
        const val FALL = "FALL"

        val ALL = arrayOf(WINTER, SPRING, SUMMER, FALL)
    }
}
